platform/golden/bullseye.pkr.hcl

```diff
@@ -53,16 +53,30 @@ variable "ssh_password" {
 }
 
 source "virtualbox-iso" "base-debian-amd64" {
+  # boot instructions/commands to send to grub/bootloader to pass preseed
   boot_command         = [
     "<esc><wait>",
-    "auto <wait>",
-    "fb=false <wait>",
-    "install <wait>", 
-    "passwd/username=packer <wait>",
-    "passwd/user-password=${var.ssh_password} <wait>",
-    "passwd/user-password-again=${var.ssh_password} <wait>",
-    "preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/${var.config_file}<wait>",
-    "<enter><wait>"
+    "c <wait>",
```

Comment:
this section, the boot mojo that defines the initial kernel command line and startup. I don't like ever touching it. I don't really understand it in detail, because we have't documented it. I understood the prior boot section, worked really hard to push all of the functionality required for that into the `pressed.cfg`.

I have spent countless hours doing magic incantations to get the bootstrap sequence right. This, right here, this part of packer, is the worst part.

It could be a little more programmatic like "bootsquenece_0:, "1, "2". this may be possible with HCL.

The general question is, can this stuff be pushed out of the kernel command line? It makes the entire image more supportable and require less work.

Reply:
all of this boot stuff is there to "manually navigate" the grub/bootloader command interface to get the instance to boot.   What I experienced was that in order to get the UEFI version of the ISO bootloader to respond, required all of the extra keyboard/locale/hostname info in the kernel boot string.

What I find is that none of this persists beyond this boot process at least, but it is certainly a pain and has limited manageability.  In the virtualbox-iso docs for Packer, there are a half dozen "examples" that are listed as effectively starting points because this is not really an API.

What I don't yet understand is why the UEFI boot loader is so different from the BIOS one relative to the debian-instsaller order of parameters (e.g. why doesn't it just read in the preseed.cfg file, and use the equivalent parameters found there!?!

Reply:
The difference that I ran into appears to be that in the EFI boot loader, the Debian-install functions do a bunch of setup _prior_ to loading the pressed.cfg via HTTP.  Basically, the network configuration (and therefore loading the pressed via URL) happens a bit later, so suddenly, you have to pass a bunch of additional configuration (keyboard, hostname, etc.) that is actually in the preseed.cfg as well.  That, and the grub boot loader’s actual commands are different (“hit c to enter config mode, pass different parameters to the Linux and initrd parameters, etc.).